src/webgpu/shader/execution/expression/binary/af_matrix_matrix_multiplication.cache.ts

```diff
@@ -0,0 +1,27 @@
+import { crc32 } from '../../../../../common/util/crc32.js';
+import { FP } from '../../../../util/floating_point.js';
+import { sparseMatrixF64Range } from '../../../../util/math.js';
+import { makeCaseCache } from '../case_cache.js';
+
+const filter_percentage = 50;
+// Cases: matKxR_matCxK
+const mat_mat_cases = ([2, 3, 4] as const)
+  .flatMap(k =>
+    ([2, 3, 4] as const).flatMap(cols =>
+      ([2, 3, 4] as const).map(rows => ({
+        [`mat${k}x${rows}_mat${cols}x${k}`]: () => {
+          return FP.abstract
+            .generateMatrixPairToMatrixCases(
+              sparseMatrixF64Range(k, rows),
+              sparseMatrixF64Range(cols, k),
+              'finite',
+              FP.abstract.multiplicationMatrixMatrixInterval
+            )
+            .filter(c => filter_percentage <= crc32(c.input.toString()) % 100);
```

Comment:
It might be preferable to specify an absolute target value instead of a percentage. The target value could then be used to calculate the relative threshold (`0xffffffff*n/count <= crc32(c.input.toString())` or something)

This could also be put in a reusable helper function.

Reply:
Done